docs/proposals/edgemesh-optimization.md

```diff
@@ -0,0 +1,81 @@
+---
+title: EdgeMesh Optimization
+authors:
+    - "@daixiang0"
+approvers:
+creation-date: 2020-06-22
+last-updated: 2020-06-22
+status:
+---
+
+# EdgeMesh Optimization
+
+## Background
+
+Now EdgeMesh works as a DNS and proxy server with some limitations:
+
+- Depend on docker0 network interface, cannot work for other CRIs
+- Only support IPV4 DNS resolver
+- Use fake ip replaced with cluster ip
+- Only support HTTP protocol proxy when apps use hostPort
+- Hack go-chassis which is a microservice framework indeed
+
+## Motivation
+This proposal addresses EdgeMesh proxy optiomization in the following ways:
+
+1. Support more DNS resolver
+2. Support more protocols
+3. Supoort cluster ip
+4. Support more port types, no only hostPort
+
+### Goals
+
+* Support multiple CRIs like docker, containerd, cri-o, etc.
+* Support IPV4 and IPV6 DNS resolver
+* Support L4 and L7 proxy
+* Support multiple protocols like HTTPS, gRPC, etc.
+* Support multiple types like containerPort, hostPort, nodePort etc.
+* Abort go-chassis
+
+### Non-goals
+
+* Cross subnet commuinication
+
+## Proposal
+
+We want EdgeMesh as lightweight as possible and support more:
+1. Support more CRIs like docker, containerd, cri-o, etc.
+2. Include more DNS resolver support as in kubernetes coreDNS
+3. Include more protocols support as in kubernetes kube-proxy
+4. Include more types support as in kubernetes kube-proxy
+
+### Use Cases
+
+* Any apps in edge side can commuinicate with each other by svc in case offline with cloud side
```

Comment:
Sorry, I'm a bit confused about this statement, it means:  any apps in the edge can communicate with each other by svc when edge nodes these apps running on are all disconnected with cloud side?

Reply:
Yep, EdgeMesh provides a DNS cache to make apps running when edge nodes offline. It would be helpful like app a and b need to communicate with each other but the node which app b run has network issue with cloud side.